arangod/Aql/OptimizerRules.cpp

```diff
@@ -2911,9 +2893,17 @@ void arangodb::aql::removeUnnecessaryCalculationsRule(Optimizer* opt,
 
   if (!toUnlink.empty()) {
     plan->unlinkNodes(toUnlink);
+    if (nodes.size() > 1) {
```

Comment:
isn't this only true if you have more remaining CalculationNodes `(nodes.size() - toUnlink.size()) > 0`

Reply:
Looks like. I made this change, and also found a bug in the if...then...else here.
Re-running it in Jenkins now...